tests/utilities/test_fetching.py

```diff
@@ -392,3 +392,45 @@ def __init__(self) -> None:
     m = InvalidModel()
     with pytest.raises(MisconfigurationException, match="is incompatible with `truncated_bptt_steps > 0`."):
         trainer.fit(m)
+
+
+@RunIf(min_gpus=1)
+def test_on_before_batch_transfer_with_unpacking(tmpdir):
```

Comment:
I think you should better check `_apply_batch_transfer_handler` or check the call count of all 3 hooks: `on_before_batch_transfer`, `transfer_batch_to_device`, `on_after_batch_transfer`.

Reply:
Yeah, let's unit-test this instead of running a `Trainer`

Reply:
Done !